iceoryx_posh/test/moduletests/test_roudi_process_introspection.cpp

```diff
@@ -169,11 +169,8 @@ TEST_F(ProcessIntrospection_test, thread)
         EXPECT_CALL(introspectionAccess.getPublisherPort().value(), offer()).Times(1);
         EXPECT_CALL(introspectionAccess.getPublisherPort().value(), sendChunk(_)).Times(Between(2, 8));
 
-        std::chrono::milliseconds& sendIntervalSleep =
-            const_cast<std::chrono::milliseconds&>(introspectionAccess.m_sendIntervalSleep);
-        sendIntervalSleep = std::chrono::milliseconds(10);
-
-        introspectionAccess.setSendInterval(10);
+        using namespace iox::units::duration_literals;
+        introspectionAccess.setSendInterval(10_ms);
```

Comment:
I guess you really like chrono :D
Well, I don't mind using chrono either. I think the idea behind duration was the easy conversion to timespec/timeval which we need for the POSIX calls. That's also the reason I use it here, since the `PeriodicTask` is build upon a `Semaphore` which expects a `units::Duration` for `timedWait`.

Reply:
+1
I would vote for replacing the `double` inside `unit::Duration` with `chrono`. In addition to it would then provide conversion operators for common POSIX time structs.

Reply:
@mossmaurice in an issue @elfenpiff suggested to use this as base
```c++
struct duration_t {
     uint64_t seconds;
     uint32_t nanoseconds;
};
```
I think it's nice for it's use case and would also suggest to do it that way

Reply:
@mossmaurice I noted this also in #190 and I think it's best discussed there or maybe in a new issue

Reply:
please use #501 for any Duration related discussions